### PR TITLE
Update dbus-rs requirement to 0.5.0

### PR DESCRIPTION
**Pull request by mulkieran**
Adapt to new API.

Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover @trgill review?

It is pretty easy to move up to new version of dbus-rs right now, so might as well.

We can take advantage of the enhancement that it provides at our leisure, but no reason not to get the bug fix in now.

**Comment by mulkieran** [comment]
Seeing if possible to make use of enhancement that attaches info to tree in a reasonably easy way...

**Comment by mulkieran** [comment]
Ok. This is ready now.

Note: I had fondly imagined that this would also allow the removal of the reference counting on the components of the DbusContext object, but I now realize that is not so.

**Comment by mulkieran** [comment]
Whoops! It is possible to do a little better.

File: src/dbus_api.rs [hunk 1]
```diff
@@ -1230,14 +1230,15 @@ fn get_base_tree<'a>(dbus_context: DbusContext) -> Tree<MTFn<TData>, TData> {
...
     let c = try!(Connection::get_private(BusType::Session));
-    c.register_name(STRATIS_BASE_SERVICE, NameFlag::ReplaceExisting as u32).unwrap();
+
+    let mut tree = get_base_tree(DbusContext::new(engine));
+    let dbus_context = tree.get_data().clone();
     try!(tree.set_registered(&c, true));
 
+    c.register_name(STRATIS_BASE_SERVICE, NameFlag::ReplaceExisting as u32).unwrap();
+
+
```
Comment (agrover) [inline src/dbus_api.rs]: extra line?

**Comment by agrover** [comment]
Yeah still need the Rc, My real hope was that we might find it useful to stick some actually objectpath-specific data in the objectpath.data() perhaps in the future.

Fix whitespace and I am 👍

**Comment by mulkieran** [comment]
@trgill review?

**Comment by trgill** [comment]
Once the conflicts are resolved - looks good.  :+1:

**Pull request merged**